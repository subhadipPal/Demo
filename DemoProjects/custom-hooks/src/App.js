import React from 'react';
import './App.css';
import { Random } from './components/Random_V1';
import { RandomV2 } from './components/Random_V2';
import { Tag } from './components/Tag_V1';
import { TagV2 } from './components/Tag_V2';

export const App = () => (
  <div>
    <h1>Random Gif App</h1>
    <div className="main-container">
      <Random />
      <Tag />
      <RandomV2 />
      <TagV2 />
    </div>
  </div>
);
