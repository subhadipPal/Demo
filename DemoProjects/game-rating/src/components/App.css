body {
  margin: 0px;
}

.App {
  min-height: 100vh;
  background-color: #f1f1f1;
  background-size: cover;
  font-family: "Times New Roman", Times, serif;
}
.App-header {
  background-color: #282c34;
  min-height: 50px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}
.App-header-text {
  font-size: 24px;
  color: white;
  margin-left: 15px;
}
.App-content {
  margin: 10px;
}
.games-grid {
  display: grid;
  grid-gap: 50px;
  grid-template-columns: auto auto auto auto;
  margin-top: 20px;
  width: 100%;
}
.grid-item {
  background-color: #fff;
  border: 1px solid #555;
  padding: 20px;
  text-align: center;
}
.game-title {
  font-size: 20px;
  font-style: italic;
}
.game-rating {
  border-top: 2px dashed #f1f1f1;
  margin-top: 10px;
  padding: 10px;
}
.field-dropdown {
  margin: 10px 0px 0px 5px;
  color: #00b1c7;
  font-size: 14px;
  outline: 0;
  border-width: 0 0 2px;
}
.dropdown-label {
  color: crimson;
}

.header-button {
  margin: 5px;
  cursor: pointer;
  border: none;
  color: #fff;
  font-size: 14px;
  text-align: center;
  padding: 10px;
  background: #555 none;
  border-radius: 5px;
}
.header-button:hover {
  background: #ccc none;
  color: #555;
}
