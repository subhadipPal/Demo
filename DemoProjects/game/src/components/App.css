body {
  background-color: black;
  display: flex;
  justify-content: center;
  width: 98vw;
  height: 98vh;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}
table {
  border-bottom: 1px solid #ccc;
}

.App {
  width: 450px;
  height: 500px;
  margin: 10px;
  background-color: #ffffff;
  box-shadow: 0px 2px 7px #eee7e7;
  -moz-box-shadow: 0px 2px 7px #eee7e7;
  -webkit-box-shadow: 0px 2px 7px #eee7e7;
  border-radius: 10px;
  -moz-border-radius: 10px;
  -webkit-border-radius: 10px;
}
.mainbody,
.App-header,
.footer {
  padding: 5px;
  background: #fff;
}

.mainbody {
  height: 100%;
  overflow-y: scroll;
}

.App-header {
  background-color: #50aadd;
  height: 40px;
  border-bottom: 1px solid #eee;
  height: 40px;
  -webkit-border-top-left-radius: 5px;
  -webkit-border-top-right-radius: 5px;
  -moz-border-radius-topleft: 5px;
  -moz-border-radius-topright: 5px;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  color: #fff;
  font-size: 20px;
}

.footer {
  background-color: whiteSmoke;
  border-top: 1px solid #ddd;
  -webkit-border-bottom-left-radius: 5px;
  -webkit-border-bottom-right-radius: 5px;
  -moz-border-radius-bottomleft: 5px;
  -moz-border-radius-bottomright: 5px;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
  overflow: hidden;
  justify-content: center;
  display: flex;
  align-items: center;
}

.button-input {
  display: flex;
  width: 56px;
  height: 56px;
  border-radius: 50%;
  background-color: #50aadd;

  color: #fff;
  text-align: center;
  float: left;
  margin: 10px 10px;
  align-items: center;
  justify-content: center;
  font-size: 17px;
  cursor: pointer;
  box-shadow: 10px;
  -webkit-box-shadow: 2px 2px 8px 0px rgba(85, 85, 85, 0.61);
  -moz-box-shadow: 2px 2px 8px 0px rgba(85, 85, 85, 0.61);
  box-shadow: 2px 2px 8px 0px rgba(85, 85, 85, 0.61);
}
.user-input {
  cursor: default !important;
  margin-left: 42%;
}
.message-row {
  align-items: center;
  justify-content: center;
}
.calculation,
.result {
  width: 90%;
  height: 40px;
  float: right;
  margin: 10px;
  -webkit-box-shadow: 0px 1px 6px 2px #eee;
  -moz-box-shadow: 0px 1px 6px 2px #eee;
  box-shadow: 0px 1px 6px 2px #eee;
  display: flex;
  align-items: center;
  justify-content: center;
}

.modalOuter {
  display: flex;
  opacity: 1;
  position: absolute;
  top: 0 !important;
  left: 0 !important;
  width: 100%;
  height: 100%;
  text-align: center;
  vertical-align: middle;
  padding: 1em;
  background-color: rgba(0, 0, 0, 0.85);
  line-height: 1;
  animation-fill-mode: both;
  animation-duration: 0.5s;
  transition: background-color 0.5s linear;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  flex-direction: column;
  -webkit-box-align: center;
  align-items: center;
  -webkit-box-pack: center;
  justify-content: center;
  user-select: none;
  will-change: opacity;
  z-index: 1000;
}

.modalInner {
  display: block;
  position: absolute;
  z-index: 1001;
  text-align: left;
  background: #fff;
  border: none;
  box-shadow: 1px 3px 3px 0 rgba(0, 0, 0, 0.2), 1px 3px 15px 2px rgba(0, 0, 0, 0.2);
  transform-origin: 50% 25%;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border-radius: 0.28571429rem;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  will-change: top, left, margin, transform, opacity;
  width: 40%;
}

.modalInner > .header {
  height: 50px;
  background-color: #50aadd;
  border-radius: 0.28571429rem 0.28571429rem 0 0;
  color: #fff;
  text-align: center;
  justify-content: center;
  align-items: center;
  display: flex;
}

.modalInner > .content {
  height: 150px;
  padding: 10px;
}

.content form {
  font-size: 1rem;
  position: relative;
  max-width: 100%;
}

.content input {
  margin: 0;
  outline: 0;
  -webkit-appearance: none;
  line-height: 2.1;
  font-size: 1em;
  padding-left: 5px;
  background: #fff;
  border: 1px solid rgba(34, 36, 38, 0.15);
  color: rgba(0, 0, 0, 0.87);
  border-radius: 0.28571429rem;
  box-shadow: 0 0 0 0 transparent inset;
  width: 99%;
}

.content button {
  cursor: pointer;
  display: inline-block;
  min-height: 1em;
  outline: 0;
  border: none;
  vertical-align: baseline;
  color: #fff;
  font-family: Lato, "Helvetica Neue", Arial, Helvetica, sans-serif;
  margin: 0 0.25em 0 0;
  padding: 0.78571429em 1.5em 0.78571429em;
  text-transform: none;
  text-shadow: none;
  font-weight: 700;
  line-height: 1em;
  font-style: normal;
  text-align: center;
  text-decoration: none;
  border-radius: 0.28571429rem;
  width: 100%;
  margin: 5px;
}

.content .start {
  background: #20c334 none;
}

.content .reset {
  background: #aaa none;
}
