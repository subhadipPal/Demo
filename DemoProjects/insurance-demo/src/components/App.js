import React, { useEffect } from "react";
import { useDispatch } from "react-redux";
import { Router, Route, Switch, Redirect } from "react-router-dom";

import history from "../history";
import LandingPage from "./home/LandingPage";
import { displayModules } from "../actions";
import Cart from "./home/Cart";

const App = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(displayModules());
  }, [dispatch]);

  return (
    <Router history={history}>
      <Switch>
        <Route path="/" exact component={LandingPage}></Route>
        <Route path="/cart" exact component={Cart}></Route>
        <Route render={() => <Redirect to="/" />} />
      </Switch>
    </Router>
  );
};

export default App;
