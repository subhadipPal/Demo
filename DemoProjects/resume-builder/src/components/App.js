import React from "react";
import { Router, Route, Switch } from "react-router-dom";

import createBrowserHistory from "../history";
import ResumeDisplay from "./resume/ResumeDisplay";
import UsernameForm from "./resume/UsernameForm";
import ErrorDisplay from "./resume/ErrorDisplay";

const App = () => {
  return (
    <div className="ui container">
      <Router history={createBrowserHistory}>
        <div>
          <Switch>
            <Route path="/" exact component={UsernameForm}></Route>
            <Route path="/resume/:username" exact component={ResumeDisplay}></Route>
            <Route path="/error/:username" exact component={ErrorDisplay}></Route>
          </Switch>
        </div>
      </Router>
    </div>
  );
};

export default App;
