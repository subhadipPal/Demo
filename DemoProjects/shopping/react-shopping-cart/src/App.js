import React, {useState} from 'react';
import data from './data.json';
import {Products} from "./components/products";

function App() {

    const [state, setState] = useState({
        products: data.products,
        size: '',
        sort: ''
    });

    return (
        <div className='grid-container'>
            <header>
                <a href='/'>React Shopping Cart</a>
            </header>
            <main>
                <div className='content'>
                    <div className='main'>
                        <Products products={state.products}   />
                    </div>
                    <div className='sidebar'>
                        Cart Items
                    </div>
                </div>
            </main>
            <footer>
                All rights is reserved.
            </footer>
        </div>
    );
}

export default App;
