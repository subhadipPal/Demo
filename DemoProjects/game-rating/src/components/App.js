import React from "react";
import { connect } from "react-redux";
import { Observable } from "rxjs/Rx";
import { range, of } from "rxjs";
import { concatMap, delay } from "rxjs/operators";

import { sortGamesByRating, randomlyRateGames } from "../actions";
import GameList from "./GameList";
import "./App.css";

class App extends React.PureComponent {
  constructor(props) {
    super(props);
    this.sortButtonAscRef = React.createRef();
    this.sortButtonDescRef = React.createRef();
  }
  componentDidMount() {
    const { sortGamesByRating, randomlyRateGames } = this.props;
    Observable.fromEvent(this.sortButtonAscRef.current, "click").subscribe(() => {
      sortGamesByRating("ASC");
    });
    Observable.fromEvent(this.sortButtonDescRef.current, "click").subscribe(() => {
      sortGamesByRating("DESC");
    });

    range(1, 10)
      .pipe(concatMap(i => of(i).pipe(delay(1000 + Math.random() * 4000))))
      .subscribe(() => {
        randomlyRateGames();
      });
  }
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <div className="App-header-text">Game Ratings</div>
        </div>
        <div className="App-content">
          <div className="list-buttons-container">
            <button className="header-button" ref={this.sortButtonAscRef} name="sortAscButton">
              Sort by Ratings (Ascending)
            </button>
            <button className="header-button" ref={this.sortButtonDescRef} name="sortDescButton">
              Sort by Ratings (Decending)
            </button>
          </div>
          <GameList />
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => {
  return {
    games: state.games
  };
};
export default connect(mapStateToProps, { sortGamesByRating, randomlyRateGames })(App);
