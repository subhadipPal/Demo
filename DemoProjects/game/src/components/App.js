import React from "react";
import { connect } from "react-redux";

import "./App.css";
import MyInput from "./MyInput";
import UserMessage from "./UserMessage";
import { updateInterface } from "../actions";
import { database } from "../firebase";
import Modal from "./Modal";

class App extends React.Component {
  componentDidMount() {
    const messagesRef = database
      .ref("messages")
      .orderByKey()
      .limitToLast(100);

    messagesRef.on("child_added", snapshot => {
      const message = { msgData: snapshot.val(), id: snapshot.key };
      this.props.updateInterface(message);
    });
  }

  render() {
    var { game } = this.props;
    return (
      <div className="App">
        <div className="App-header">Game (Player:{game.user})</div>
        <div className="mainbody">
          <div className="content">
            <UserMessage></UserMessage>
          </div>
        </div>
        <div className="footer">
          <MyInput></MyInput>
        </div>
        <Modal></Modal>
      </div>
    );
  }
}
const mapStateToProps = state => {
  return {
    game: state.game
  };
};
export default connect(mapStateToProps, { updateInterface })(App);
