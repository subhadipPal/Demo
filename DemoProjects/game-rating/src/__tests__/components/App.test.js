import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import App from "../../components/App";

const mockStore = configureStore([]);

describe("App Component test", () => {
  let store = {};
  let wrappedComponent = {};

  beforeEach(() => {
    store = mockStore({
      games: []
    });
    store.dispatch = jest.fn();
    wrappedComponent = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  it("should render with given state from Redux store", () => {
    expect(wrappedComponent.toJSON()).toMatchSnapshot();
  });
  it("should have a App", () => {
    expect(wrappedComponent.root.findByProps({ className: "App" })).toBeDefined();
  });
});
