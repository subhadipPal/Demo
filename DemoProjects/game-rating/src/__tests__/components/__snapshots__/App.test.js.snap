// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component test should render with given state from Redux store 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <div
      className="App-header-text"
    >
      Game Ratings
    </div>
  </div>
  <div
    className="App-content"
  >
    <div
      className="list-buttons-container"
    >
      <button
        className="header-button"
        name="sortAscButton"
      >
        Sort by Ratings (Ascending)
      </button>
      <button
        className="header-button"
        name="sortDescButton"
      >
        Sort by Ratings (Decending)
      </button>
    </div>
    Loading...
  </div>
</div>
`;
